### ISSUE TITLE
Generation of node application with docker

### ISSUE THREAD

**Issue by abrahamsaanchez** [nrwl/nx#14813]
### Current Behavior

I'm generating an application with the `@nrwl/node:application` setting the `docker` option to true.

The serve target are looking for a `production` configuration in the build target which is not defined.

### Expected Behavior

The configuration should be defined.

### Steps to Reproduce

1. Generate an application with `@nrwl/node:application` and docker set to true

### Nx Report

```shell
Node : 18.13.0
OS   : linux x64
npm  : 8.19.3

nx : 15.6.3
@nrwl/angular : 15.6.3
@nrwl/cypress : 15.6.3
@nrwl/detox : Not Found
@nrwl/devkit : 15.6.3
@nrwl/esbuild : 15.6.3
@nrwl/eslint-plugin-nx : 15.6.3
@nrwl/expo : Not Found
@nrwl/express : Not Found
@nrwl/jest : 15.6.3
@nrwl/js : 15.6.3
@nrwl/linter : 15.6.3
@nrwl/nest : Not Found
@nrwl/next : Not Found
@nrwl/node : 15.6.3
@nrwl/nx-cloud : Not Found
@nrwl/nx-plugin : Not Found
@nrwl/react : Not Found
@nrwl/react-native : Not Found
@nrwl/rollup : Not Found
@nrwl/schematics : Not Found
@nrwl/storybook : Not Found
@nrwl/web : Not Found
@nrwl/webpack : 15.6.3
@nrwl/workspace : 15.6.3
@nrwl/vite : Not Found
typescript : 4.9.4
---------------------------------------
Local workspace plugins:
---------------------------------------
Community plugins:
```

**Comment by ndcunningham** [comment]
I am not sure what you mean could you explain further?

All targets defined should work on a fresh application.

**Comment by abrahamsaanchez** [comment]
It's seems to be fixed in 15.9.2
